staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -333,7 +333,7 @@ func (r *Request) Name(resourceName string) *Request {
 		return r
 	}
 	if len(resourceName) == 0 {
-		r.err = fmt.Errorf("resource name may not be empty")
+		r.err = errors.NewBadRequest("resource name may not be empty")
```

Comment:
we shouldn't just be changing this one return type ... all the errors in this file are client-side, not server-side, which is interesting ... 

<img width="903" height="471" alt="image" src="https://github.com/user-attachments/assets/defed97d-0050-4c2c-8b44-5d42b214f7f1" />

I'm not sure if we should always be wrapping in NewBadRequest before assigning to r.err, or should be wrapping r.err in NewBadRequest before returning it